### PR TITLE
Keep registrants from brute-forcing SMS OTP verification

### PR DESCRIPTION
**Pull request by rjkip**
- [ ] Cherry-pick into develop?

### CODE REVIEW

File: app/Resources/translations/validators.nl_NL.xliff [hunk 1]
```diff
@@ -212,6 +212,11 @@
         <source>ss.prove_phone_possession.send_sms_challenge_failed</source>
         <target>Het versturen van de code per SMS is mislukt.</target>
       </trans-unit>
+      <trans-unit id="f355c0b1302e6ed1a9301637306cb312b6feb4eb" resname="ss.prove_phone_possession.too_many_attempts">
+        <jms:reference-file line="24">Resources/views/translations.twig</jms:reference-file>
+        <source>ss.prove_phone_possession.too_many_attempts</source>
+        <target>U heeft de limiet van drie pogingen bereikt; u kunt geen codes meer verifiëren. Neem contact op met uw helpdesk of probeer het later nog eens.</target>
```
Comment (DRvanR) [inline app/Resources/translations/validators.nl_NL.xliff]: `drie` != `ten` :wink:

File: src/Surfnet/StepupSelfService/SelfServiceBundle/Service/SmsSecondFactor/OtpVerification.php [hunk 2]
```diff
@@ -23,6 +23,7 @@
     const STATUS_NO_MATCH = 0;
     const STATUS_MATCH_EXPIRED = 1;
     const STATUS_FOUND_MATCH = 2;
+    const STATUS_TOO_MANY_ATTEMPTS = 3;
```
Comment (DRvanR) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Service/SmsSecondFactor/OtpVerification.php]: not configurable? (Check with SURF)

File: src/Surfnet/StepupSelfService/SelfServiceBundle/Service/SmsSecondFactor/SmsVerificationState.php [hunk 3]
```diff
@@ -86,6 +97,12 @@ public function requestNewOtp($phoneNumber)
      */
     public function verify($userOtp)
     {
+        if ($this->verificationAttemptsMade >= self::MAXIMUM_VERIFICATION_ATTEMPTS) {
+            return OtpVerification::tooManyAttempts();
```
Comment (DRvanR) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Service/SmsSecondFactor/SmsVerificationState.php]: Maybe I'm missing something, but I can't see this working as the count is reset to `0` every instantiation and it doesn't seem to be stored anywhere? (the test checking this reuses the same instance, which is not the case in consecutive requests)

File: app/Resources/translations/validators.nl_NL.xliff (hunk 1 above)
Comment (rjkip) [inline app/Resources/translations/validators.nl_NL.xliff]: :')

File: src/Surfnet/StepupSelfService/SelfServiceBundle/Service/SmsSecondFactor/SmsVerificationState.php (hunk 3 above)
Comment (rjkip) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Service/SmsSecondFactor/SmsVerificationState.php]: Ceci, n'est pas une probleme: https://github.com/SURFnet/Stepup-SelfService/blob/8f64a3f3f0f012b2e49b95aa5cf3e30bb3f5a039/src/Surfnet/StepupSelfService/SelfServiceBundle/Service/SmsSecondFactor/SessionSmsVerificationStateHandler.php#L76

Comment (DRvanR) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Service/SmsSecondFactor/SmsVerificationState.php]: tsenks

**Pull request closed without merging**